docs/hardening_guide.md

```diff
@@ -0,0 +1,405 @@
+```
+
+Run sysctl -p `/etc/sysctl.d/90-kubelet.conf` to enable the settings.
+
+## Configure Etcd User and Group
+
+A user account and group for the etcd service is required to be setup prior to installing RKE2. The uid and gid for the etcd user will be used in the RKE2 config.yml.
+
+### Create the Etcd User and Group
+
+To create the etcd group run the following console commands.
+
+The commands below use 52034 for uid and gid are for example purposes. Any valid unused uid or gid could also be used in lieu of 52034.
+
+```sh
+groupadd --gid 52034 etcd
+useradd --comment "etcd service account" --uid 52034 --gid 52034 etcd
+```
+
+Update the RKE config.yml with the uid and gid of the etcd user:
+
+```yaml
+services:
+  etcd:
+    gid: 52034
+    uid: 52034
+```
+
+## Set automountServiceAccountToken to false for Default Service Accounts
+
+Kubernetes provides a default service account which is used by cluster workloads where no specific service account is assigned to the pod. Where access to the Kubernetes API from a pod is required, a specific service account should be created for that pod, and rights granted to that service account. The default service account should be configured such that it does not provide a service account token and does not have any explicit rights assignments.
+
+For each namespace including default and kube-system on a standard RKE install the default service account must include this value:
+
+```yaml
+automountServiceAccountToken: false
+```
+
+Save the following yaml to a file called account_update.yaml
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: default
+automountServiceAccountToken: false
+```
+
+Create a bash script file called account_update.sh. Be sure to chmod +x account_update.sh so the script has execute permissions.
+
+```sh
+#!/bin/bash -e
+
+for namespace in $(kubectl get namespaces -A -o json | jq -r '.items[].metadata.name'); do
+    kubectl patch serviceaccount default -n ${namespace} -p "$(cat account_update.yaml)"
+done
+```
+
+### Ensure that all Namespaces have Network Policies defined
+
+Running different applications on the same Kubernetes cluster creates a risk of one compromised application attacking a neighboring application. Network segmentation is important to ensure that containers can communicate only with those they are supposed to. A network policy is a specification of how selections of pods are allowed to communicate with each other and other network endpoints.
+
+Network Policies are namespace scoped. When a network policy is introduced to a given namespace, all traffic not allowed by the policy is denied. However, if there are no network policies in a namespace all traffic will be allowed into and out of the pods in that namespace. To enforce network policies, a CNI (container network interface) plugin must be enabled. This guide uses canal to provide the policy enforcement. Additional information about CNI providers can be found here
```

Comment:
Link?: "Additional information about CNI providers can be found here" missing link